Core/include/Acts/Seeding/SeedfinderUtils.hpp

```diff
@@ -0,0 +1,240 @@
+#pragma once
+
+#include "Acts/Utilities/KDRange.hpp"
+
+template<typename SP>
+Acts::LinCircle _lincircle(
+  const SP & middle,
+  const SP & other,
+  bool bottom
+) {
+  float xM = middle.x();
+  float yM = middle.y();
+  float zM = middle.z();
+  float rM = middle.radius();
+  float varianceZM = middle.varianceZ();
+  float varianceRM = middle.varianceR();
+  float cosPhiM = xM / rM;
+  float sinPhiM = yM / rM;
+
+  float deltaX = other.x() - xM;
+  float deltaY = other.y() - yM;
+  float deltaZ = other.z() - zM;
+
+  float x = deltaX * cosPhiM + deltaY * sinPhiM;
+  float y = deltaY * cosPhiM - deltaX * sinPhiM;
+
+  float iDeltaR2 = 1. / (deltaX * deltaX + deltaY * deltaY);
+  float iDeltaR = std::sqrt(iDeltaR2);
+
+  int bottomFactor = 1 * (int(!bottom)) - 1 * (int(bottom));
+
+  float cot_theta = deltaZ * iDeltaR * bottomFactor;
+
+  Acts::LinCircle l;
+  l.cotTheta = cot_theta;
+
+  l.Zo = zM - rM * cot_theta;
+  l.iDeltaR = iDeltaR;
+
+  l.U = x * iDeltaR2;
+  l.V = y * iDeltaR2;
+
+  l.Er = ((varianceZM + other.varianceZ()) +
+          (cot_theta * cot_theta) * (varianceRM + other.varianceR())) *
+          iDeltaR2;
+  return l;
+}
+
+template<typename SP>
+Acts::LinCircle lcBotMid(
+  const SP & bottom,
+  const SP & middle
+) {
+  return _lincircle(middle, bottom, true);
+}
+
+template<typename SP>
+Acts::LinCircle lcMidTop(
+  const SP & middle,
+  const SP & top
+) {
+  return _lincircle(middle, top, false);
+}
+
+template<typename SP>
```

Comment:
Where do these numbers come? Is this from the original Seeding code - I think that's super tailored to ATLAS.

@robertlangenberg - can you comment on that?